internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -2325,3 +2343,19 @@ func getClusterDeploymentAdminKubeConfigSecretName(cd *hivev1.ClusterDeployment)
 	}
 	return fmt.Sprintf(adminKubeConfigStringTemplate, cd.Name)
 }
+
+// processMirrorRegistryConfig retrieves the mirror registry configuration from the referenced ConfigMap
+func (r *ClusterDeploymentsReconciler) processMirrorRegistryConfig(ctx context.Context, log logrus.FieldLogger, clusterInstall *hiveext.AgentClusterInstall) (*hiveext.MirrorRegistryConfiguration, error) {
+	mirrorRegistryConfiguration, userTomlConfigMap, err := mirrorregistries.ProcessMirrorRegistryConfig(ctx, log, r.Client, clusterInstall.Spec.MirrorRegistryRef)
```

Comment:
I'd prefer we read the configmap data in the controllers package here and keep the knowledge of the kube API out of the internal packages.

Reply:
Not sure I understood what you meant here

Reply:
You're sending a configmap reference to the `mirrorregistries` package, but I would expect that package to take the data as input rather than the configmap.

For example `ProcessMirrorRegistryConfig` should also be usable from the REST API functions (if we were ever to implement them)